packages/sdk/react-client/src/echo/SpaceContext.tsx

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright 2020 DXOS.org
+//
+
+import React, { createContext, FC, PropsWithChildren, useContext, useEffect, useState } from 'react';
+
+import { Space } from '@dxos/client';
+
+import { useClient } from '../client';
+import { useSpaces } from './useSpaces';
+
+export type SpaceContextProps = {
+  space?: Space;
+  setSpace: (space?: Space) => void;
+};
+
+export const SpaceContext: React.Context<SpaceContextProps> = createContext<SpaceContextProps>({ setSpace: () => {} });
+
+export const SpaceProvider: FC<PropsWithChildren<{}>> = ({ children }) => {
+  const client = useClient();
+  const spaces = useSpaces();
+  const [space, setSpace] = useState<Space>();
+
+  useEffect(() => {
+    return client.shell?.spaceUpdate.on((spaceKey) => {
```

Comment:
Does the shell only support a single space at a time? (That would be bad).
Also, I don't think we should expose `shell` from client, although it would add complexity to inject a listener.

Reply:
The shell currently supports selecting only a single space at a time in line with how all of our apps currently work. It ties into the new space context to set the current space for an app. Next on the todo list for shell is to add views for the current space and a space switcher.